insonmnia/hub/server.go

```diff
@@ -165,105 +163,18 @@ func (h *Hub) Info(ctx context.Context, request *pb.ID) (*pb.InfoReply, error) {
 	return resp, nil
 }
 
-func decodePortBinding(v string) (string, string, error) {
-	mapping := strings.Split(v, "/")
-	if len(mapping) != 2 {
-		return "", "", errors.New("failed to decode Docker port mapping")
-	}
-
-	return mapping[0], mapping[1], nil
-}
-
-type extRoute struct {
+type routeMapping struct {
 	containerPort string
 	route         *route
 }
 
-type minerFilter func(miner *MinerCtx, requirements *pb.TaskRequirements) (bool, error)
-
-// ExactMatchFilter checks for exact match.
-//
-// Returns true if there are no miners specified in the requirements. In that
-// case we must apply hardware filtering for discovering miners that can start
-// the task.
-// Otherwise only specified miners become targets to start the task.
-func exactMatchFilter(miner *MinerCtx, requirements *pb.TaskRequirements) (bool, error) {
-	if len(requirements.GetMiners()) == 0 {
-		return true, nil
-	}
-
-	for _, minerID := range requirements.GetMiners() {
-		if minerID == miner.ID() {
-			return true, nil
-		}
-	}
-
-	return false, nil
-}
-
-func resourcesFilter(miner *MinerCtx, requirements *pb.TaskRequirements) (bool, error) {
-	resources := requirements.GetResources()
-	if resources == nil {
-		return false, status.Errorf(codes.InvalidArgument, "resources section is required")
-	}
-
-	cpuCount := resources.GetCPUCores()
-	memoryCount := resources.GetMaxMemory()
-	var gpuCount = 0
-	if resources.GetGPUSupport() {
-		gpuCount = -1
-	}
-
-	var usage = resource.NewResources(int(cpuCount), int64(memoryCount), gpuCount)
-	if err := miner.PollConsume(&usage); err != nil {
-		return false, err
-	}
-
-	return true, nil
-}
-
-func (h *Hub) selectMiner(request *pb.HubStartTaskRequest) (*MinerCtx, error) {
-	requirements := request.GetRequirements()
-	if requirements == nil {
-		return nil, status.Errorf(codes.InvalidArgument, "missing requirements")
-	}
-
-	// Filter out miners that aren't met the requirements.
-	h.mu.Lock()
-	defer h.mu.Unlock()
-	miners := []*MinerCtx{}
-	for _, miner := range h.miners {
-		ok, err := h.applyFilters(miner, requirements)
-		if err != nil {
-			return nil, err
-		}
-
-		if ok {
-			miners = append(miners, miner)
-		}
-	}
-
-	// Select random miner from the list.
-	if len(miners) == 0 {
-		return nil, status.Errorf(codes.NotFound, "failed to find miner to match specified requirements")
-	}
-
-	return miners[rand.Int()%len(miners)], nil
-}
-
-func (h *Hub) applyFilters(miner *MinerCtx, req *pb.TaskRequirements) (bool, error) {
-	for _, filter := range h.filters {
-		ok, err := filter(miner, req)
-		if err != nil {
-			return false, err
-		}
-
-		if !ok {
-			return false, nil
-		}
+func (h *Hub) onRequest(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp interface{}, err error) {
```

Comment:
Strange diff. Did you change anything in onRequest?

Reply:
Nope.